### PR TITLE
Add new metric for consensus time.

### LINKED ISSUE #5725: Create metric for consensus time
Needed to help debug the freeze issue: https://github.com/hashgraph/hedera-services/issues/5684

Create a metric that shows latest consensus timestamp. Perhaps as seconds or milliseconds since the epoch.

Also, create a companion metric that shows the difference between consensus timestamp and wall clock timestamp while we are at it.

### PR DESCRIPTION
**Pull request by cody-littley**
Closes #5725

### CODE REVIEW

**Review by edward-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ConsensusHandlingMetrics.java [hunk 1]
```diff
@@ -123,4 +143,15 @@ public CycleTimingStat getNewSignedStateCycleStat() {
     public void recordEventsPerRound(final int numEvents) {
         avgEventsPerRound.update(numEvents);
     }
+
+    /**
+     * Records the consensus time.
+     *
+     * @param consensusTime
+     * 		the consensus time of the most recently handled round
```
Comment (poulok) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ConsensusHandlingMetrics.java:151]:
```suggestion
     * 		the consensus time of the last event of most recently handled round
```

**Review by alittley — DISMISSED** [review]

**Review by poulok — CHANGES_REQUESTED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/ConsensusRoundHandler.java [hunk 2]
```diff
@@ -316,6 +316,9 @@ private void applyConsensusRoundToState(final ConsensusRound round) throws Inter
 
         consensusTimingStat.setTimePoint(1);
 
+        if (round.getEventCount() > 0) {
+            consensusHandlingMetrics.recordConsensusTime(round.getLastEvent().getConsensusTimestamp());
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/ConsensusRoundHandler.java:320]: fixed

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ConsensusHandlingMetrics.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ConsensusHandlingMetrics.java:151]: Updated javadoc to read `the consensus time of the last transaction in the round that is currently being handled`.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ConsensusHandlingMetrics.java [hunk 3]
```diff
@@ -123,4 +143,15 @@ public CycleTimingStat getNewSignedStateCycleStat() {
...
+     */
+    public void recordConsensusTime(final Instant consensusTime) {
+        this.consensusTime.set(consensusTime.toEpochMilli());
+        consensusTimeDeviation.set(consensusTime.toEpochMilli() - Instant.now().toEpochMilli());
```
[the remark this answers is not in the document]
Comment (cody-littley) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ConsensusHandlingMetrics.java:155]: updated to use Time

**Review by alittley — APPROVED** [review]

**Review by poulok — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Review by alittley — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]

**Pull request merged**